src/snowcli/cli/common/sql_execution.py

```diff
@@ -3,14 +3,28 @@
 import logging
 from contextlib import contextmanager
 from functools import cached_property
-from textwrap import dedent
+from io import StringIO
+from textwrap import dedent, indent
+from typing import Iterable
 
 from click import ClickException
 from snowcli.cli.common.cli_global_context import cli_context
-from snowflake.connector.cursor import DictCursor
+from snowflake.connector.cursor import DictCursor, SnowflakeCursor
 from snowflake.connector.errors import ProgrammingError
 
 
+class _SnowcliCursor(SnowflakeCursor):
+    def execute(self, *args, **kwargs):
+        """This allows us to show failing query to users."""
+        # TODO: Use exception annotation once we support Python 3.11
+        try:
+            super().execute(*args, **kwargs)
+        except Exception:
+            print("Query:")
```

Comment:
It looks like python's `print`, not click's `print`. Did you consider click's `print` and check it? Maybe we can get some nice formatting. I am just curious, not asking to change it.